src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -1,19 +1,57 @@
 package core.basesyntax.impl;
 
 import core.basesyntax.Storage;
+import java.lang.reflect.Array;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_ARRAY_LENGTH = 10;
+    private K[] keyArray;
+    private V[] valueArray;
+    private int firstEmptyIndex;
+
+    public StorageImpl() {
+        firstEmptyIndex = 0;
+        keyArray = (K[]) Array.newInstance(Object.class, MAX_ARRAY_LENGTH);
+        valueArray = (V[]) Array.newInstance(Object.class, MAX_ARRAY_LENGTH);
+    }
+
     @Override
     public void put(K key, V value) {
+        int containsIndex = contains(firstEmptyIndex, key);
+        if (containsIndex != -1) {
+            replace(containsIndex, value);
+        } else if (firstEmptyIndex < MAX_ARRAY_LENGTH) {
+            replace(firstEmptyIndex, key, value);
```

Comment:
I think replacing the value will be enough 
```suggestion
            replace(firstEmptyIndex, value);
```

Reply:
I remove firstEmptyIndex from paramethers, but i need key in case when i put a new pair in storage. I have to add key to the end of keyArray